src/components/video/VideoJS.jsx

```diff
@@ -11,59 +11,115 @@ window.videojs = videojs;
 // eslint-disable-next-line import/no-extraneous-dependencies
 require('videojs-vjstranscribe');
 
+function useTranscripts({ player, customOptions }) {
+  const shouldUseTranscripts = customOptions?.showTranscripts && customOptions?.transcriptUrls
+  const [isLoading, setIsLoading] = useState(shouldUseTranscripts);
+  const [textTracks, setTextTracks] = useState([]);
+  const [transcriptUrl, setTranscriptUrl] = useState(null);
+
+  useEffect(() => {
+    const fetchFn = async () => {
+      setIsLoading(true);
+      if (shouldUseTranscripts) {
+        try {
+          const result = await fetchAndAddTranscripts(customOptions.transcriptUrls, player);
+          setTextTracks(result);
+          // We are only catering to English transcripts for now as we don't have the option to change
+          // the transcript language yet.
+          if (result.en) {
+            setTranscriptUrl(result.en); 
+          }
+        } catch (error) {
+          logError(`Error fetching transcripts for player: ${error}`);
+        } finally {
+          setIsLoading(false);
+        }
+      }
+    };
+    fetchFn();
+  }, []);
+
+  return {
+    textTracks,
+    transcriptUrl,
+    isLoading,
+  };
+}
+
 const VideoJS = ({ options, onReady, customOptions }) => {
   const videoRef = useRef(null);
   const playerRef = useRef(null);
 
+  const transcriptsData = useTranscripts({
+    player: playerRef.current,
+    customOptions,
+  });
+
   useEffect(() => {
     // Make sure Video.js player is only initialized once
     if (!playerRef.current) {
+      if (transcriptsData.isLoading) {
+        // while async transcripts data is loading, don't initialize the player
+        return;
+      }
+      
       // The Video.js player needs to be _inside_ the component el for React 18 Strict Mode.
       const videoElement = document.createElement('video-js');
 
       videoElement.classList.add('vjs-big-play-centered');
       videoRef.current.appendChild(videoElement);
 
-      // eslint-disable-next-line no-multi-assign
-      const player = playerRef.current = videojs(videoElement, options, () => {
+      const transformedPlayerOptions = {
+        ...options,
+        plugins: {
+          vjstranscribe: {
+            urls: transcriptsData.transcriptUrl ? [transcriptsData.transcriptUrl] : [],
```

Comment:
[inform] the plugin's `this.on(player, 'ready', function() { ... })` callback function was never executing for me locally until I pass into the `videojs(...)` initialization options:

```jsx
{
  plugins: {
    vjstranscribe: { urls: [...] },
  },
}
```

This suggests the `vjstranscribe` plugin requires upfront setting of the transcript URL during `videojs` initialization  as since the plugin does not appear to properly handle remote text tracks added via `addRemoteTextTrack` asynchronously.

In short, the crux of the fix is to fetch the transcript URL prior to `videojs(...)` initialization.

Reply:
@adamstankiewicz One suggestion is to include the `vjstranscribe` plugin only if `showTranscripts` is set to true. This way, the plugin will not be initialized when the video player is rendering on course previews or the course about page, where transcripts are not needed.

NIT:
```
const transformedPlayerOptions = {
  ...options,
  plugins: customOptions?.showTranscripts
    ? {
        vjstranscribe: {
          urls: transcriptsData.transcriptUrl ? [transcriptsData.transcriptUrl] : [],
        },
      }
    : {},
};
```

Reply:
Good suggestion! I will include an update to account for this suggestion in the next push addressing your feedback.